src/main/java/org/broadinstitute/hellbender/dev/utils/ReadRecord.java

```
@@ -0,0 +1,274 @@
+    if (null != supplementaryAlignment) {
+      this.setSupplementaryAlignmentFlag(supplementaryAlignment);
+    }
+    String alignedSequence = read.getAlignedSequence();
+    if (null != alignedSequence) {
+      this.setReadString(alignedSequence);
+    }
+    List<Integer> alignedQuality = read.getAlignedQuality();
+    if (null != alignedQuality) {
+      this.setBaseQualities(integerListToByteArray(alignedQuality));
+    }
+    if (read.getNumberReads()>1) {
+      // 0x1: template having multiple segments in sequencing
+      this.setReadPairedFlag(true);
+      // 0x80:  the last segment in the template
+      this.setSecondOfPairFlag(read.getReadNumber() + 1 == read.getNumberReads());
+      // 0x8: next segment in the template unmapped
+      this.setMateUnmappedFlag(read.getNextMatePosition()==null);
+    }
+    Position nextMatePosition = read.getNextMatePosition();
+    if (null != nextMatePosition) {
+      String referenceName = nextMatePosition.getReferenceName();
+      if (null != referenceName) {
+        this.setMateReferenceName(referenceName);
+      }
+      Long position = nextMatePosition.getPosition();
+      if (null != position) {
+        // convert from 0-based to 1-based
+        this.setMateAlignmentStart(position.intValue()+1);
+      }
+      Boolean reverseStrand = nextMatePosition.getReverseStrand();
+      if (null != reverseStrand) {
+        this.setMateNegativeStrandFlag(reverseStrand);
+      }
+    }
+    Map<String, List<String>> info = read.getInfo();
+    if (null != info) {
+      for (Map.Entry<String, List<String>> entry : info.entrySet()) {
+        String key = entry.getKey();
+        List<String> value = entry.getValue();
+        try {
+          Tag tag = Tag.valueOf(key);
+          this.setAttribute(
+              key, null == tag ? Iterables.getOnlyElement(value) : tag.convert(value));
+        } catch (IllegalArgumentException x) {
+          // X,Y, or Z tag. Not sure what to do.
```

Don't swallow exceptions without a good explanation. I'd suggest throwing an appropriate exception instead (even if it's in the constructor).  Not sure if IllegalArgumentException is appropriate here, or if a custom one is warranted.